jest/private/bazel_haste_map.cjs

```diff
@@ -0,0 +1,106 @@
+const { join, dirname, extname } = require("path");
+const { readFileSync } = require("fs");
+
+// to require jest-haste-map we need to hop from jest-cli => @jest/core => jest-haste-map in the virtual store
+const jestCliPackage = dirname(require.resolve("jest-cli/package.json"));
+const jestConfigPackage = dirname(
+  require.resolve(join(jestCliPackage, "../@jest/core/package.json")),
+);
+const HasteMap = require(
+  join(jestConfigPackage, "../../jest-haste-map"),
+).default;
+const fastPath = require(
+  join(jestConfigPackage, "../../jest-haste-map/build/lib/fast_path.js"),
+);
+
+// The path to the rules_jest files list file
+const WORKSPACE_RUNFILES = join(
+  process.env.TEST_SRCDIR,
+  process.env.TEST_WORKSPACE,
+);
+const RULES_JEST_FILES = join(
+  WORKSPACE_RUNFILES,
+  global.BAZEL_FILELIST_SHORT_PATH,
+);
+
+/**
+ * Extend the standard jest HasteMap to use rules_jest
+ */
+module.exports = class BazelHasteMap extends HasteMap {
+  constructor(options) {
+    super(options);
+
+    // Override the jest HasteMap._crawl() to not invoke `find` or `fs.*`
+    this._crawl = this.bazelCrawl.bind(this);
+
+    // Persisting within rules_jest is pointless
+    this._persist = function () {};
```

Comment:
Not worthless if there were a worker mode or if sandboxing were disabled for the test. Not saying I'm advocating for persistence, but the comment indicated that maybe those scenarios hadn't been thought of

Reply:
I've expanded this comment a bunch to explain it more. While it's often "pointless" it also crashes atm due to the class we are extending no longer being invoked by its factory method.